### PR TITLE
example: fix custom 802.15.4 channel definition

### PR DESCRIPTION
**Pull request by aabadie**
While playing with gnrc_networking example and setting a custom channel, I noticed that the changes from #5977 were not applied.

### CODE REVIEW

**Comment by miri64** [comment]
Weird was sure I included it... Since this might be problematic with the default channel definition I mark it as a bug fix that needs a backport (after it was merged to master).

File: examples/gnrc_networking/Makefile [hunk 1]
```diff
@@ -34,9 +34,19 @@ USEMODULE += netstats_l2
 USEMODULE += netstats_ipv6
 USEMODULE += netstats_rpl
 
-# Set a custom 802.15.4 channel if needed
-DEFAULT_CHANNEL ?= 26
-CFLAGS += -DDEFAULT_CHANNEL=$(DEFAULT_CHANNEL)
+# Set a custom channel if needed
+ifeq (,$(filter cc110x,$(USEMODULE)))     # radio is cc110x sub-GHz
```
Comment (aabadie) [inline examples/gnrc_networking/Makefile]: Well, after trying again, this doesn't work actually. `$(USEMODULE)` only contains the modules defined above in this particular Makefile. The modules dependencies is processed __after__ so '$(USEMODULE)' never contains `cc110x` and the code always end up in this block, even if I try to compile for a samr21-xpro (which uses an `at86rf2xx` driver).

I'm not an expert in Makefile so I may have missed something.

Comment (miri64) [inline examples/gnrc_networking/Makefile]: Ahh, sorry yes! You need to move it below l. 63 (after the include of `$(RIOTBASE)/Makefile.include`)

Comment (miri64) [inline examples/gnrc_networking/Makefile]: (the reason is that the dependencies are pulled in in this include ;-))

**Comment by aabadie** [comment]
> You need to move it below l. 63

Doesn't make any difference. I think the `DEFAULT_CHANNEL` is broken in the other examples as well (just tested with microcoap_server, same problem)

**Comment by aabadie** [comment]
You probably initially tested on a `msba2` board ?

**Comment by miri64** [comment]
Mhhhhh .... no. I don't have any of that boards. Now that I think about it I might not have tested this properly. Will think about a fix and PR it against this one, if you don't mind.

**Comment by aabadie** [comment]
> Will think about a fix and PR it against this one, if you don't mind.

Thanks, that would help. You'll have to split the drivers dependency discovering in a separate Makefile and include it before calling the default channel initialization block. I don't know if such a Makefile already exists.

**Comment by miri64** [comment]
I summon @smlng into the discussion ;-)

**Comment by miri64** [comment]
See https://github.com/aabadie/RIOT/pull/2. The problem was just a `ifeq`/`ifneq`-switch-up.

**Comment by aabadie** [comment]
@miri64, ok. Is it ok if I adapt this PR with the ifneq ?

**Comment by aabadie** [comment]
(I'll also change the other Makefiles)

**Comment by aabadie** [comment]
@miri64, I tested, squashed and rebased. This is all good now. Thanks !

File: examples/default/Makefile [hunk 2]
```diff
@@ -72,14 +72,14 @@ endif
 include $(RIOTBASE)/Makefile.include
 
 # Set a custom channel if needed
-ifeq (,$(filter cc110x,$(USEMODULE)))     # radio is cc110x sub-GHz
+ifneq (,$(filter cc110x,$(USEMODULE)))          # radio is cc110x sub-GHz
```
Comment (aabadie) [inline examples/default/Makefile]: @miri64, thinking of it : would it make sense to move this block somewhere in [Makefile.include](https://github.com/RIOT-OS/RIOT/blob/master/Makefile.include), maybe [here](https://github.com/RIOT-OS/RIOT/blob/master/Makefile.include#L206)

Then no need to duplicate this code in all applications ?

**Comment by miri64** [comment]
Needs backport

**Comment by aabadie** [comment]
> Needs backport

Who should do that ?

**Comment by miri64** [comment]
Preferably you ;-)

**Comment by aabadie** [comment]
> Preferably you ;-)

Ok, should I create a PR in 2016.10-RC1 from my fork ? (what's the best practice for that ?)

**Comment by aabadie** [comment]
issued #6040

**Pull request merged**